File: BinarySearch/Python/tests/test_binarysearch.py
import unittest
from src.binarysearch import binary_search

class TestBinarysearch(unittest.TestCase):

	def test_binarysearch_simple(self):
		array = [1,2,3,4,5]

		self.assertEqual(binary_search(array, 1), 0)
		self.assertEqual(binary_search(array, 2), 1)
		self.assertEqual(binary_search(array, 3), 2)
		self.assertEqual(binary_search(array, 4), 3)
		self.assertEqual(binary_search(array, 5), 4)

	def test_binarysearch(self):
		array = [1, 25, 43, 50, 156, 999]

		self.assertEqual(binary_search(array, 1), 0)
		self.assertEqual(binary_search(array, 25), 1)
		self.assertEqual(binary_search(array, 43), 2)
		self.assertEqual(binary_search(array, 50), 3)
		self.assertEqual(binary_search(array, 156), 4)
		self.assertEqual(binary_search(array, 999), 5)
		self.assertEqual(binary_search(array, 157), -1)

if __name__ == '__main__':
	unittest.main()

File: BinarySearch/Java/Tests/TestBinarySearch.java
import static org.junit.Assert.*;
import org.junit.*;
import BinarySearch.*;

public class TestBinarySearch {

	@Test
	public void TestBinarySearchSimple() {
		int[] array = {1, 2, 3, 4, 5};

		assertEquals(BinarySearch.search(array, 1), 0);
		assertEquals(BinarySearch.search(array, 2), 1);
		assertEquals(BinarySearch.search(array, 3), 2);
		assertEquals(BinarySearch.search(array, 4), 3);
		assertEquals(BinarySearch.search(array, 5), 4);
		assertEquals(BinarySearch.search(array, 6), -1);
		assertEquals(BinarySearch.search(array, -1), -1);
	}

	@Test
	public void TestBinarySearch() {
		int[] array = new int[100];

		for (int i = 0; i < 100; i++) {
			array[i] = i;
		}

		assertEquals(BinarySearch.search(array, 50), 50);
		assertEquals(BinarySearch.search(array, 35), 35);
		assertEquals(BinarySearch.search(array, 9), 9);
		assertEquals(BinarySearch.search(array, 34), 34);
		assertEquals(BinarySearch.search(array, -5), -1);
	}

		@Test
		public void TestBinarySearchSingle() {
			int[] array = {1};

			assertEquals(BinarySearch.search(array, 1), 0);
			assertEquals(BinarySearch.search(array, 11), -1);
			assertEquals(BinarySearch.search(array, 10), -1);
			assertEquals(BinarySearch.search(array, -1), -1);
			assertEquals(BinarySearch.search(array, -5), -1);
		}

		@Test
		public void TestBinarySearchEmpty() {
			int[] array = {};

			assertEquals(BinarySearch.search(array, 1), -1);
			assertEquals(BinarySearch.search(array, 11), -1);
			assertEquals(BinarySearch.search(array, 10), -1);
			assertEquals(BinarySearch.search(array, -1), -1);
			assertEquals(BinarySearch.search(array, -5), -1);
		}

}
